text/3550-new-range.md

```diff
@@ -0,0 +1,441 @@
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Change the range operators `a..b`, `a..`, and `a..=b` to resolve to new types `ops::range::Range`, `ops::range::RangeFrom`, and `ops::range::RangeInclusive` in Edition 2024. These new types will not implement `Iterator`, instead implementing `Copy` and `IntoIterator`.
+
+# Motivation
+[motivation]: #motivation
+
+The current iterable range types ([`Range`], [`RangeFrom`], [`RangeInclusive`]) implement `Iterator` directly. This is now widely considered to be a mistake, because it makes implementing `Copy` for those types hazardous due to how the two traits interact.
+
+```rust
+for x in it.take(3) {  // a *copy* of the iterator is used here
+    // ..
+}
+
+match it.next() {  // the original iterator (not advanced) is used here
+    // ..
+}
+```
+
+[`Range`]: https://doc.rust-lang.org/stable/core/ops/struct.Range.html
+[`RangeFrom`]: https://doc.rust-lang.org/stable/core/ops/struct.RangeFrom.html
+[`RangeInclusive`]: https://doc.rust-lang.org/stable/core/ops/struct.RangeInclusive.html
+
+However, there is considerable demand for `Copy` range types for multiple reasons:
+- ergonomic use without needing explicit `.clone()`s or rewriting the `a..b` syntax repeatedly
+- use in `Copy` types (currently people work around this by using a tuple instead)
+
+Another primary motivation is the extra size of `RangeInclusive`. It uses an extra `bool` field to keep track of when the upper bound has been yielded by the iterator, but this extra size is useless when the type is not used as an iterator.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Rust has several different types of "range" syntax, including the following:
+
+- `a..b` denotes a range from `a` (inclusive) to `b` (exclusive). It resolves to the type `std::ops::range::Range`.  
+  The iterator for `Range` will yield values from `a` (inclusive) to `b` (exclusive) in steps of one.
+
+- `a..=b` denotes a range from `a` (inclusive) to `b` (inclusive). It resolve to the type `std::ops::range::RangeInclusive`.
+  The iterator for `RangeInclusive` will yield values from `a` (inclusive) to `b` (inclusive) in steps of one.
+
+- `a..` denotes a range from `a` (inclusive) with no upper bound. It resolves to the type `std::ops::range::RangeFrom`.  
+  The iterator for `RangeFrom` will yield values starting with `a` and increasing in steps of one.
+
+These types implement the `IntoIterator` trait, enabling their use directly in a `for` loop:
+```rust
+for n in 0..5 {
+    // `n` = 0, 1, 2, 3, 4
+}
+```
+
+All range types are `Copy` when the bounds are `Copy`, allowing easy reuse:
+```rust
+let range = 0..5;
+
+if a_slice[range].contains(x) {
+    // ...
+}
+if b_slice[range].contains(y) {
+    // ...
+}
+```
+
+For convenience, several commonly-used methods from `Iterator` are present as inherent functions on the range types:
```

Comment:
At some point the RFC should say which "several" methods it refers to here (or determining that list should explicitly be made an unresolved question).

Reply:
I do explicitly defer this to be determined after the RFC is accepted. It's the first thing under Unresolved questions

> The set of inherent methods copied from `Iterator` present on the new range types is left open for the **libs-api** team to decide after this proposal is accepted and before stabilization.

Reply:
Oh, I carefully read through all the items in there and missed the part before. Please format everything consistently and use an item for each unresolved question.